### DISCUSSION TITLE
Build app for macos fails

### DISCUSSION THREAD

**Discussion by caspahouzer** [appcelerator/titanium_mobile#12640] [Q&A]
I tried to build my app for `macos`. I set macos to true in the tiapp.xml.

Build it with `ti build -p ios -T macos` fails in 
```
[ERROR] ** BUILD FAILED **
[ERROR] The following build commands failed:
[ERROR]         Ld /Users/sebastianklaus/Documents/Apps/MyApp/build/iphone/build/Intermediates/MyApp/Debug-maccatalyst/MyApp/Objects-normal/x86_64/Binary/MyApp normal x86_64
[ERROR] (1 failure)
```

This error does not help a lot in finding out where the problem is. Anyone can help?

**Comment by Topener** [comment]
Do you have any modules? Perhaps certain modules don't have `x86_64` support

**Reply by caspahouzer** [reply]
I checked all modules. They all have `x86_64`

**Reply by Topener** [reply]
Can you try to lower log level to debug, and see what is last performed before this error hits? This is just the final step and might not actually indicate the problem.

**Reply by jquick-axway** [reply]
This build error can happen with a module whose native library does not support Mac. For example, the "facebook" module will cause this same build error because Facebook's native library does not support Mac.

So, I recommend that you comment out all of your modules from your "tiapp.xml" and then rebuild for Mac. If it builds successfully, then that proves one of the modules isn't support on Mac. You'll then have to work backwards adding a module back in one at a time until you find the module that's not supported on Mac. Unfortunately, I'm not aware of a better way to do it at the moment. Not even debug logging will reveal the culprit in this case.

**Reply by vijaysingh-axway** [reply]
@caspahouzer To check if any module supports mac or not -
1. Check if it has .xcframework or not. If not it means it does not support the mac. Location of .xcframework is 
2. If it has .xcframework. Go inside .xcframework, check if it has ios-arm64_x86_64-maccatalyst  folder or not. If not it means this module does not support mac.
3. If the module does not support mac, comment out it in tiapp.xml. 
4. I general, if a module is made using third party framework do not have mac support.
